File: 백준/Silver/15649. N과 M （1）/N과 M （1）.py
def backtrack(n, m, visited, sequence):
    if len(sequence) == m:
        print(' '.join(map(str, sequence)))
        return

    for i in range(1, n + 1):
        if not visited[i]:
            visited[i] = True
            sequence.append(i)
            backtrack(n, m, visited, sequence)
            sequence.pop()
            visited[i] = False

n, m = map(int, input().split())
visited = [False] * (n + 1)
backtrack(n, m, visited, [])

File: 백준/Silver/15649. N과 M （1）/N과 M （1）.js
const fs = require("fs");
const input = fs.readFileSync(0, "utf8").trim().split(" ");

const n = +input[0];
const m = +input[1];
const visited = Array(n + 1).fill(false);
const seq = [];
const output = [];

function backtrack(depth) {
  if (depth === m) {
    output.push(seq.join(" "));
    return;
  }

  for (let num = 1; num <= n; num++) {
    if (visited[num]) continue;

    visited[num] = true;
    seq.push(num);

    backtrack(depth + 1);

    seq.pop();
    visited[num] = false;
  }
}

backtrack(0);
console.log(output.join("\n"));